TemplateMethod/EmailVerification.cs

```diff
@@ -0,0 +1,41 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Epam.NetMentoring.Adapter.TemplateMethod
+{
+    public abstract class EmailVerification
+    {
+
+        private readonly Template _template;
+
+        public EmailVerification(Template template)
+        {
+            _template = template;
+        }
+
+        public void Verification()
+        {
+            CheckUserAccess(_template);
```

Comment:
How do you break execution flow, no one method throws or return code which can be used for making any decision?

Reply:
what about error reporting/logging?

Reply:
ok